### PR TITLE
Fix DeepSeek MoE expert stacked regressions

### LINKED ISSUE #42965: DeepSeek MoE expert decode test fails PCC after timeout fix
This tracks a separate numerical regression from [#42941](https://github.com/tenstorrent/tt-metal/issues/42941), which was covering the DeepSeek multi-host timeout failure.

The first failing test in the current DeepSeek multi-host CI run is:
- Workflow/job: [DeepSeek V3 Multi-Host Tests / deepseekv3-tests (dual)](https://github.com/tenstorrent/tt-metal/actions/runs/24817924884/job/72636851605)
- Test: `models/demos/deepseek_v3/tests/test_moe_experts.py::test_forward_pass[model.layers.3.mlp.experts.0-255-real-decode-32-1]`

On remote branch `codex/deepseek-moe-stacked-ref-setup` (commit [`229c2a8b2b4`](https://github.com/tenstorrent/tt-metal/commit/229c2a8b2b4ebe16accb00c96ccc17a9043f02b2)), the timeout regression is fixed for this repro: the test now completes in about 94 seconds instead of timing out at 300 seconds.

However, once the timeout is removed, the same repro still fails the numerical check:
- Observed PCC: `0.975726`
- Required PCC: `0.98`

This suggests the timeout/setup regression and the remaining PCC miss should be tracked separately. The timeout fix branch appears to expose a pre-existing or otherwise separate accuracy issue that was previously masked by the timeout.

Repro command used for the CI-style dual-host path:
```bash
eval "$(./tests/scripts/multihost/setup_shared_venv.sh --activate)"

export TT_METAL_HOME=$PWD
export PYTHONPATH=$PWD
export ARCH_NAME=wormhole_b0
export RANK_BINDING_YAML="tests/tt_metal/distributed/config/dual_galaxy_rank_bindings.yaml"
export MESH_GRAPH_DESCRIPTOR="tt_metal/fabric/mesh_graph_descriptors/dual_galaxy_mesh_graph_descriptor.textproto"
export HOSTS="$(awk '!/^#/ && NF {print $1}' /etc/mpirun/hostfile | head -n 2 | paste -sd,)"
export RANKFILE=/etc/mpirun/rankfile
export MPI_ARGS="--host $HOSTS --map-by rankfile:file=$RANKFILE --bind-to none --output-filename logs/mpi_job"
export TCP_INTERFACE=cnx1
export DEEPSEEK_V3_HF_MODEL="/mnt/MLPerf/tt_dnn-models/deepseek-ai/DeepSeek-R1-0528-dequantized"
export DEEPSEEK_V3_CACHE="/mnt/MLPerf/tt_dnn-models/deepseek-ai/DeepSeek-R1-0528-Cache/CI"
export MESH_DEVICE=DUAL
mkdir -p logs generated/artifacts

tt-run --tcp-interface "$TCP_INTERFACE" \
  --mesh-graph-descriptor "$MESH_GRAPH_DESCRIPTOR" \
  --hosts "$HOSTS" \
  pytest -svvv \
  'models/demos/deepseek_v3/tests/test_moe_experts.py::test_forward_pass[model.layers.3.mlp.experts.0-255-real-decode-32-1]'
```

### PR DESCRIPTION
**Pull request by yieldthought**
## Summary
- fix stacked DeepSeek MoE reference setup so real-weight expert tests slice `experts_stacked.*` directly instead of repeatedly materializing legacy per-expert aliases
- restore pre-quantization transpose semantics for MoE expert weights so BF4/BF8 conversion matches the last known-good behavior
- add focused regression coverage for stacked alias caching, layer-filter visibility, BSPM stacked inputs, and stacked reference assembly

## Root Cause
- `#42150` switched DeepSeek MoE to stacked checkpoints, but the real-weight test reference path still iterated legacy per-expert aliases and timed out on stacked tensors
- the same change also moved the expert-weight transpose from conversion time to runtime via `transpose_b=True`, which changed the BF4/BF8 conversion path and likely caused the remaining PCC regression after the timeout fix

## Issue
Fixes #42965

## Testing
- `pre-commit run --files models/demos/deepseek_v3/tests/test_bspm_demo.py models/demos/deepseek_v3/tests/test_hf_model_utils.py models/demos/deepseek_v3/tests/test_lazy_state_dict.py models/demos/deepseek_v3/tests/test_moe_experts.py models/demos/deepseek_v3/tt/experts.py models/demos/deepseek_v3/utils/lazy_state_dict.py`
- `python3 -m py_compile models/demos/deepseek_v3/tt/experts.py models/demos/deepseek_v3/utils/lazy_state_dict.py models/demos/deepseek_v3/tests/test_moe_experts.py models/demos/deepseek_v3/tests/test_lazy_state_dict.py models/demos/deepseek_v3/tests/test_hf_model_utils.py models/demos/deepseek_v3/tests/test_bspm_demo.py`
- `git diff --check`
- current-head CI: [all-static-checks run 24850727010](https://github.com/tenstorrent/tt-metal/actions/runs/24850727010), [galaxy-deepseek-tests run 24850777785](https://github.com/tenstorrent/tt-metal/actions/runs/24850777785)
- local `pytest` selectors could not be run on this machine because the Python environment is missing `pytest`, `torch`, `safetensors`, `loguru`, and `transformers`

## CI Badge
[All static checks (yieldthought/42965-deepseek-moe-pcc-fix)](https://github.com/tenstorrent/tt-metal/actions/workflows/all-static-checks.yaml?query=branch%3Ayieldthought%2F42965-deepseek-moe-pcc-fix)
[Galaxy DeepSeek tests (yieldthought/42965-deepseek-moe-pcc-fix)](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-deepseek-tests.yaml?query=branch%3Ayieldthought%2F42965-deepseek-moe-pcc-fix)

### CODE REVIEW

**Comment by yieldthought** [comment]
Fixed as part of #42966
